docs/references/architecture/adr-115-predictable-block-times.md

```diff
@@ -0,0 +1,113 @@
+# ADR 115: Predictable Block Times
+
+## Changelog
+
+ - April 30 2024: Created by @melekes
+
+## Status
+
+**Proposed**
+
+## Context
+
+Let's say you're an ABCI application developer and you want to have
+predictable block times: 1 block each 6s. How would you do that?
+
+You found out that proposing a block and voting on it takes roughly 2s in your
+network. You change `timeout_commit` (_how long a node waits for additional
+precommits after committing a block, before starting on the next height_) from
+1s to 4s in the default config and ship your app with that modified config. Do
+you have predictable block times now?
+
+No - you don't. The block time will be around 6s, but it won't be exactly 6s. A
+validator could set `timeout_commit`  to 0s because it's a local parameter,
+which is not enforced by Comet. That means whenever this validator proposes a
+block, the overall block time will be 2s (not 6s!).
+
+There are other reasons why the block time might not be predictable:
+
+1. CometBFT can go into multiple rounds of consensus, which will increase the
+   block time.
+2. Network latency.
+3. Clock drifts.
+
+By now, many teams (Celestia, Berachain, Osmosis, etc.) expressed their
+interest in either a) having predictable block times OR b) having the ability
+to alter the next block time.
+
+In case of Osmosis, committing a big block might take longer than usual. They
+could increase the block time for big blocks to give the state machine some
+extra time to finish execution if that was possible.
+
+In case of Berachain, they want to have a predictable block time to match
+Ethereum's slots, which are equal to 12s.
+
+## Decision
+
+Remove `timeout_commit` in favor of a new field in `FinalizeBlockResponse`:
+`next_block_delay`. This field's semantics stays essentially the same: delay
+between this block and the time when the next block is proposed.
```

Comment:
What you mean by "between this block". When this interval starts? When `FinalizeBlock` returns?